test-poc/framework/src/main/java/org/keycloak/test/framework/oauth/OAuthCallbackServer.java

```diff
@@ -0,0 +1,60 @@
+package org.keycloak.test.framework.oauth;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedList;
+import java.util.List;
+
+class OAuthCallbackServer {
+
+    private final HttpServer httpServer;
+    private final OAuthCallbackHandler callbackHandler;
+    private final URI redirectionUri;
+
+    public OAuthCallbackServer() {
```

Comment:
this is a replacement for the Oauth provider?

Reply:
What is the OAuth provider?

Reply:
Quick summary from what we discussed in chat - the old testsuite has an oauth client, but that is a bit messy (and relies on a test app deployed to KC server as a realm resource provider for callbacks) so didn't want to introduce it here (at least for now). Nimbus stuff is just a temporary thing, and not quite sure what we will use in the long term for the new testsuite.